### PR TITLE
Enable `prettier` globally

### PR DESCRIPTION
**Pull request by tlrobinson**
We've had [prettier](https://prettier.io/) enabled on `metabase-lib` and a few other parts of the app for awhile, and I'm generally a fan. It makes the whole code base a bit more consistent style wise, and lets you not have to think about formatting. Sometimes I just bang out code with gibberish styling then hit control-option-F to clean it up (or just set your editor to run it on save)

They fixed the issue I had with ternary formatting, so I think it's time to consider enabling it for the whole app. They also added support for CSS, so it seems logical to enable it for that too while we're at it.

Here are some steps that I think should work for merging existing PRs that modify frontend files (I haven't fully tested this so check the results carefully):

```bash
# Merge the commit immediately prior to running prettier globally (includes upgrading and enabling prettier globally):
git merge 5cd9b2d77
# Manually resolve and fix conflicts if necessary, then commit.
# Save a list of changed files between that commit and your branch pre-prettier as a sanity check
git diff --name-only 5cd9b2d77 > before-prettier
# Update the dependencies and run `prettier` on your branch:
yarn
yarn run prettier
# Commit the results
git commit -a -m "Run prettier on BRANCH NAME"
# Cherry-pick some fixes
git cherry-pick f311d26be
# Then merge with same commit
git merge -Xours f311d26be
# Save a list of changed files between that commit and your newly merged branch, then diff it with the previously saved list to make sure there aren't any unexpectedly changed files
git diff --name-only f311d26be > after-prettier
diff before-prettier after-prettier
# Merge with master:
git merge origin/master
# Run lint/flow/tests
yarn run lint && yarn run flow && yarn run test
```

### CODE REVIEW

File: frontend/src/metabase/admin/datamodel/selectors.js [hunk 1]
```diff
@@ -72,21 +83,22 @@ export const revisionHistorySelectors = createSelector(
...
-export const getEditingTable          = (state, props) => state.admin.datamodel.editingTable;
-export const getEditingDatabase       = (state, props) => state.admin.datamodel.editingDatabase;
-
+export const getDatabases = (state, props) => state.admin.datamodel.databases;
+export const getDatabaseIdfields = (state, props) =>
+    state.admin.datamodel.idfields;
+export const getEditingTable = (state, props) =>
+    state.admin.datamodel.editingTable;
+export const getEditingDatabase = (state, props) =>
+    state.admin.datamodel.editingDatabase;
```
Comment (camsaul) [inline frontend/src/metabase/admin/datamodel/selectors.js]: I liked the old formatting better here

File: frontend/src/metabase/admin/datamodel/components/database/MetadataHeader.jsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ export default class MetadataHeader extends Component {
         databases: PropTypes.array.isRequired,
         selectDatabase: PropTypes.func.isRequired,
         isShowingSchema: PropTypes.bool.isRequired,
-        toggleShowSchema: PropTypes.func.isRequired,
+        toggleShowSchema: PropTypes.func.isRequired
```
Comment (camsaul) [inline frontend/src/metabase/admin/datamodel/components/database/MetadataHeader.jsx]: it's removing all the hanging commas!

**Comment by camsaul** [comment]
Overall this LGTM and I think it's a net win but unfortunately it looks like there are some places where it second-guesses meticulously hand-formatted code and makes it objectively less pretty.

**Comment by kdoh** [comment]
Very down for this.

File: frontend/src/metabase/admin/datamodel/components/database/MetadataHeader.jsx (hunk 2 above)
Comment (salsakran) [reply frontend/src/metabase/admin/datamodel/components/database/MetadataHeader.jsx]: Dura Lex Sed Lex

Comment (tlrobinson) [reply frontend/src/metabase/admin/datamodel/components/database/MetadataHeader.jsx]: we could enable this option https://prettier.io/docs/en/options.html#trailing-commas

**Comment by tlrobinson** [comment]
@metabase/core-developers I've updated this and come up with instructions for merging existing branches. I've also enabled trailing commas and changed tabs to be 2 spaces.

Anyone opposed to merging this? Speak now or forever hold your peace.

**Pull request merged**